File: 2017/FizzBuzz/FizzBuzz/RefreshingCache/FizzBuzz.cs
namespace RefreshingCache
{
    public class FizzBuzz
    {
        private const string Fizz = "Fizz";
        private const string Buzz = "Buzz";

        public string Process(int number)
        {
            if (number.IsDividableBy(15))
            {
                return Fizz + Buzz;
            }

            if (number.IsDividableBy(5))
            {
                return Buzz;
            }

            if (number.IsDividableBy(3))
            {
                return Fizz;
            }

            return number.ToString();
        }
    }

    public static class Int32Extensions
    {
        public static  bool IsDividableBy(this int number, int divider)
        {
            return number % divider == 0;
        }
    }
}

File: 2017/Koshelev/fizzBuzz/fizzBuzz.js
Number.prototype.isDividable = function (number){
	return this % number == 0;
}

class FizzBuzz{
	
	constructor(options){
		this._register = options || {
			'3': 'Fizz',
			'5': 'Buzz'
		};
		
		if (!this._register['3']){
			this._register['3'] = 'Fizz';
		}

		if (!this._register['5'])
			this._register['5'] = 'Buzz';
		
	}

	process(number){
		let result = '';
		let register = this._register;

		for (let key in register){
			if (number.isDividable(key))
			result += register[key];
		}

		if (!result)
			return number.toString();
		return result;
	}

	add(options){
		for(let key in options){
			this._register[key] = options[key];
		}
	}
}

module.exports = FizzBuzz;